src/sentry/static/sentry/app/components/activity/noteInput.jsx

```diff
@@ -238,20 +253,50 @@ const NoteInput = createReactClass({
     }
   },
 
+  mentionableUsers() {
+    let {memberList, sessionUser} = this.props;
+    return _.uniqBy(memberList, ({id}) => id)
+      .filter(member => sessionUser.id !== member.id)
+      .map(member => ({
+        id: buildUserId(member.id),
+        display: member.name,
+        email: member.email,
+      }));
+  },
+
+  mentionableTeams() {
+    let {group} = this.props;
+
+    return _.uniqBy(TeamStore.getAll(), ({id}) => id)
```

Comment:
This and `mentionableUsers` need to go into state otherwise you'll be calling these methods on every render (and that happens on every keystroke I'm guessing).

Reply:
👍

Reply:
good catch, that was slightly hand wavy of me because i thought about it when i moved it but didn't consider that it would be each keystroke and could be a large number of members being mapped over